ocdata/test/flag_anomaly.py

```diff
@@ -1,104 +1,120 @@
+
+        Returns:
+            (bool): True if there is desorption, otherwise False
         """
-        adsorbate_atoms_idx = [idx for idx, tag in enumerate(self.atoms_tag) if tag==2]
-        surface_atoms_idx = [idx for idx, tag in enumerate(self.atoms_tag) if tag==1]
-        adslab_struct = AseAtomsAdaptor.get_structure(self.final_atoms)
+        adsorbate_atoms_idx = [
+            idx for idx, tag in enumerate(self.atoms_tag) if tag == 2
+        ]
+        surface_atoms_idx = [idx for idx, tag in enumerate(self.atoms_tag) if tag != 2]
+        final_connectivity = self._get_connectivity(
+            self.final_atoms, self.desorption_cutoff_multiplier
+        )
 
-        vnn = VoronoiNN(allow_pathological=True, tol=0.8, cutoff=6)
         for idx in adsorbate_atoms_idx:
-            neighbors = adslab_struct.get_neighbors(adslab_struct[idx], neighbor_thres)
-            surface_neighbors = [n.index for n in neighbors if n.index in surface_atoms_idx]
-            if len(surface_neighbors) != 0:
+            if sum(final_connectivity[idx][surface_atoms_idx]) >= 1:
                 return False
         return True
 
-    def _find_max_movement(self, init_atoms, final_atoms):
-        '''
-        Given ase.Atoms objects, find the furthest distance that any single atom in
-        a set of atoms traveled (in Angstroms)
-        Args:
-                init_atoms      `ase.Atoms` of the structure in its initial state
-                final_atoms     `ase.Atoms` of the structure in its final state
-        Returns:
-                max_movement    A float indicating the further movement of any single atom
-                                before and after relaxation (in Angstroms)
-        '''
-        # Calculate the distances for each atom
-        distances = final_atoms.positions - init_atoms.positions
-
-        # Reduce the distances in case atoms wrapped around (the minimum image convention)
-        _, movements = find_mic(distances, final_atoms.cell, final_atoms.pbc)
-        max_movement = max(movements)
-        return max_movement
-
-    def _get_connectivity(self, atoms):
+    def _get_connectivity(self, atoms, cutoff_multiplier=1.0):
         """
         Generate the connectivity of an atoms obj.
+
         Args:
-                    atoms      An `ase.Atoms` object
+            atoms (ase.Atoms): object which will have its connectivity considered
+            cutoff_multiplier (float, optional): cushion for small atom movements when assessing
+                atom connectivity
+
         Returns:
-                    matrix     The connectivity matrix of the atoms object.
+            (np.ndarray): The connectivity matrix of the atoms object.
         """
-        cutoff = natural_cutoffs(atoms)
-        neighborList = neighborlist.NeighborList(cutoff, self_interaction=False, bothways=True)
+        cutoff = list(np.array(natural_cutoffs(atoms)) * cutoff_multiplier)
+        neighborList = neighborlist.NeighborList(
```

Comment:
Maybe change this variable name to nbr_list (lots of variations of neighborlist that are part of the ase method), ase_neighbor_list, or neighbor_list.

Reply:
that makes sense!